DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/bridging/test_bridging_learning_address.py

```diff
@@ -0,0 +1,122 @@
+    ports = tgen_dev.links_dict[device_host_name][1]
+    traffic_duration = 10
+
+    out = await IpLink.add(
+        input_data=[{device_host_name: [{"device": bridge, "type": "bridge"}]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that bridge created.\n{out}"
+
+    out = await IpLink.set(
+        input_data=[{device_host_name:  [
+            {"device": port, "master": "br0", "operstate": "up"} for port in ports]},
+            {"device": bridge, "operstate": "up"}])
+    err_msg = f"Verify that bridge, bridge entities set to 'UP' state and links enslaved to bridge.\n{out}"
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    out = await BridgeLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "learning": True, "flood": False} for port in ports]}])
+    err_msg = f"Verify that entities set to learning 'ON' and flooding 'OFF' state.\n{out}"
+    assert out[0][device_host_name]["rc"] == 0, err_msg
+
+    address_map = (
+        # swp port, tg port,     tg ip,      gw,         plen
+        (ports[0], tg_ports[0], "11.0.0.1", "11.0.0.4", 24),
+        (ports[1], tg_ports[1], "11.0.0.2", "11.0.0.3", 24),
+        (ports[2], tg_ports[2], "11.0.0.3", "11.0.0.2", 24),
+        (ports[3], tg_ports[3], "11.0.0.4", "11.0.0.1", 24),
+    )
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {"ixp": port, "ip": ip, "gw": gw, "plen": plen}
+        for _, port, ip, gw, plen in address_map
+    )
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    streams = {
+        f"bridge_{dst + 1}": {
+            "ip_source": dev_groups[tg_ports[src]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[dst]][0]["name"],
+            "srcMac": f"aa:bb:cc:dd:ee:1{dst + 1}",
+            "dstMac": "ff:ff:ff:ff:ff:ff",
+            "type": "raw",
+            "protocol": "802.1Q",
+        } for src, dst in ((3, 0), (2, 1), (1, 2), (0, 3))
+    }
+
+    await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(traffic_duration)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # check the traffic stats
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    for row in stats.Rows:
+        assert tgen_utils_get_loss(row) != 100.000, f'Failed>Loss percent: {row["Loss %"]}'
+
+    out = await BridgeFdb.show(input_data=[{device_host_name: [{"cmd_options": "-j"}]}],
+                               parse_output=True)
+
+    devices = out[0][device_host_name]["parsed_output"]
+    data = []
+    for dev in devices:
+        data.append(dev.get("mac", None))
+
+    test_data_in = ["aa:bb:cc:dd:ee:11", "aa:bb:cc:dd:ee:12",
+                    "aa:bb:cc:dd:ee:13", "aa:bb:cc:dd:ee:14"]
+    for addr in test_data_in:
+        err_msg = f"Verify that source macs have been learned.\n{out}"
+        assert addr in test_data_in, err_msg
```

Comment:
What are you verifying here? You are not using the collected before `data` list.